Sources/Mailgun/MailgunClient.swift

```diff
@@ -0,0 +1,117 @@
+//
+//  MailgunClient.swift
+//  Mailgun
+//
+//  Created by Anthony Castelli on 11/14/16.
+//
+//
+
+import Console
+import Vapor
+import HTTP
+import Foundation
+import TurnstileCrypto
+import SMTP
+import Mail
+
+/**
+ Mailgun client
+ */
+public final class MailgunClient {
+    
+    static var defaultApiKey: String?
+    static var defaultDomain: String?
+    static var defaultClient: ClientProtocol.Type?
+    
+    var domain: String
+    var apiKey: String
+    var client: ClientProtocol
+    
+    public init(clientProtocol: ClientProtocol.Type, domain: String, apiKey: String) throws {
+        self.domain = domain
+        self.apiKey = apiKey
+        self.client = try clientProtocol.make(scheme: "https", host: "api.mailgun.net")
+    }
+    
+    public func send(_ emails: [MailgunEmail]) throws {
+        try emails.forEach { email in
+            let boundary = "vapor.mailgun.package.\(URandom().secureToken)"
```

Comment:
I'm not sure. I haven't messed around with it in 1.5.x, but it was partly implemented before and I never got it to work properly. Will have another look though.